homeassistant/components/sensorpush/__init__.py

```diff
@@ -0,0 +1,60 @@
+"""The Govee Bluetooth integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.update_coordinator import (
+    BluetoothDataUpdateCoordinator,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN
+from .parser import parse_sensorpush_from_discovery_data
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Govee Bluetooth from a config entry."""
+    address = entry.unique_id
+    assert address is not None
+
+    @callback
+    def _async_parse_sensorpush_device(
+        service_info: bluetooth.BluetoothServiceInfo, change: bluetooth.BluetoothChange
+    ) -> dict[str, Any] | None:
+        """Subscribe to bluetooth changes."""
+        if data := parse_sensorpush_from_discovery_data(
+            service_info.name,
+            service_info.manufacturer_data,
+        ):
+            data["rssi"] = service_info.rssi
+            return data
+        return None
```

Comment:
This is still too much boiler plate

RSSI should be automatic and disabled by default